src/cmdstan/stansummary_helper.hpp

```diff
@@ -463,129 +407,51 @@ void write_header(const std::vector<std::string> &header,
+            && j >= stats.cols() - 3)
+          continue;  // don't report ESS or Rhat for sampler state
         std::cout.setf(col_formats(j), std::ios::floatfield);
-        *out << std::setprecision(
-            compute_precision(params(i, j), sig_figs,
-                              col_formats(j) == std::ios_base::scientific))
-             << std::setw(col_widths(j)) << params(i, j);
+        *out << std::setprecision(compute_precision(
+            stats(i, j), sig_figs, col_formats(j) == std::ios_base::scientific))
+             << std::setw(col_widths(j)) << stats(i, j);
       }
     }
     *out << std::endl;
-    i++;
-  }
-}
-
-/**
- * Output statistics for a set of parameters
- * either as fixed-width text columns or in csv format.
- * Containers are re-ordered as first-index-major order
- *
- * @param in set of samples from one or more chains
- * @param in matrix of statistics
- * @param in vector of output column widths
- * @param in vector of output column formats
- * @param in size of longest parameter name - (width of 1st output column)
- * @param in significant digits required
- * @param in index of first column in chains object
- * @param in output format flag:  true for csv; false for plain text
- * @param in output stream
- */
-void write_all_model_params(const stan::mcmc::chains<> &chains,
-                            const Eigen::MatrixXd &params,
-                            const Eigen::VectorXi &col_widths,
-                            const Eigen::Matrix<std::ios_base::fmtflags,
-                                                Eigen::Dynamic, 1> &col_formats,
-                            int max_name_length, int sig_figs,
-                            int params_start_col, bool as_csv,
-                            std::ostream *out) {
-  for (int i = 0, i_chains = params_start_col; i < params.rows();
-       ++i, ++i_chains) {
-    if (!is_container(chains.param_name(i_chains))) {
-      if (as_csv) {
-        *out << "\"" << chains.param_name(i_chains) << "\"";
-        for (int j = 0; j < params.cols(); j++) {
-          *out << "," << params(i, j);
-        }
-      } else {
-        *out << std::setw(max_name_length + 1) << std::left
-             << chains.param_name(i_chains);
-        *out << std::right;
-        for (int j = 0; j < params.cols(); j++) {
-          out->setf(col_formats(j), std::ios::floatfield);
-          *out << std::setprecision(
-              compute_precision(params(i, j), sig_figs,
-                                col_formats(j) == std::ios_base::scientific))
-               << std::setw(col_widths(j)) << params(i, j);
-        }
-      }
-      *out << std::endl;
-    } else {
-      // container object columns in csv are last-index-major order
-      // output as first-index-major order
-      std::vector<int> dims = dimensions(chains, i_chains);
-      std::vector<int> index(dims.size(), 1);
-      int max = 1;
```

Comment:
I can't tell if this code ended up somewhere else, but I wanted to point out that this ordering behavior did end up changing in this PR:

Previously:

```
$ ./bin/stansummary ./src/test/interface/matrix_output.csv 
Inference for Stan model: issue_342_model
1 chains: each with iter=(10); warmup=(0); thin=(1); 10 iterations saved.

Warmup took 0.23 seconds
Sampling took 0.0018 seconds

                Mean   MCSE  StdDev       5%   50%   95%  N_Eff   N_Eff/s  R_hat

lp__             -15    1.2     3.7 -2.6e+01   -15   -13     10   5.7e+03   0.92
accept_stat__   0.69   -nan    0.30     0.20  0.86   1.0    -45  -2.6e+04   0.92
stepsize__      0.89    nan    0.00     0.89  0.89  0.89    nan       nan    nan
treedepth__     0.90   0.10    0.32     0.00   1.0   1.0     10   5.7e+03    nan

x[1]            0.61   -nan    0.22  2.8e-01  0.67  0.98    -48  -2.7e+04   0.89
x[2]            0.48  0.051    0.16  1.8e-01  0.48  0.77     10   5.7e+03   0.99
y[1,1]          0.43   0.12    0.39  2.3e-02  0.17  0.93     10   5.7e+03    1.1
y[1,2]          0.60  0.076    0.24  2.5e-01  0.71  0.88     10   5.7e+03   0.90
y[1,3]          0.47   -nan    0.33  4.5e-02  0.71  0.94    -92  -5.2e+04   0.90
y[2,1]          0.51   -nan    0.27  1.7e-01  0.49  0.93   -354  -2.0e+05   0.91
y[2,2]          0.56  0.087    0.28  1.6e-01  0.65  0.99     10   5.7e+03   0.94
y[2,3]          0.52  0.100    0.32  1.9e-03  0.65  0.93     10   5.7e+03    1.2
```

Now:
```
$ ./bin/stansummary ./src/test/interface/matrix_output.csv 
Inference for Stan model: issue_342_model
1 chains: each with iter=10; warmup=1000; thin=1; 10 iterations saved.

Warmup took 0.23 seconds
Sampling took 0.0018 seconds

                Mean   MCSE   StdDev   MAD     5%   50%   95%  ESS_bulk  ESS_tail  R_hat

lp__             -15    1.2  3.7e+00  0.78    -21   -15   -13       5.0       5.0    1.3
accept_stat__   0.69  0.096  3.0e-01  0.25   0.21  0.78  0.99
stepsize__      0.89    nan  1.2e-16  0.00   0.89  0.89  0.89
treedepth__     0.90   0.10  3.2e-01  0.00   0.45   1.0   1.0

x[1]            0.61  0.069  2.2e-01  0.11   0.28  0.67  0.87       5.0       5.0   0.91
x[2]            0.48  0.051  1.6e-01  0.16   0.26  0.45  0.69       5.0       5.0    1.0
y[1,1]          0.43   0.12  3.9e-01  0.15  0.059  0.16  0.93       5.0       5.0    1.4
y[2,1]          0.51  0.087  2.7e-01  0.38   0.17  0.49  0.90       5.0       5.0    1.4
y[1,2]          0.60  0.076  2.4e-01  0.18   0.25  0.66  0.85       5.0       5.0    1.1
y[2,2]          0.56  0.087  2.8e-01  0.31   0.23  0.53  0.93       5.0       5.0   0.92
y[1,3]          0.47   0.10  3.3e-01  0.39  0.068  0.48  0.86       5.0       5.0   0.99
y[2,3]          0.52  0.100  3.2e-01  0.21  0.036  0.54  0.93       5.0       5.0    1.3
```

Reply:
I see - the `else` loop beginning on line 551 (old version) - re-orders output for container objects.   ugh.

Reply:
I have rewritten the code so that multi-dim container params are reported in row-major order, as before.

This interacts badly with the "--include_params" feature, in 2 ways:

1. "--include_params" requires specifying all elements of a container variable explicitly, and given that the user has requested them in that order, that's what they get.
2. there was an existing bug in that with "-i", you couldn't specify multiple input files following a requested param - the arg parser would get confused. 

```
 ~/.cmdstan/cmdstan-2.35.0> bin/stansummary -i theta bern[1-4].csv
--include_param: Unrecognized parameter(s): 'bern1[csv]' 'bern2[csv]' 'bern3[csv]' 
```
To fix this, I added a `-f` flag for input files, and a unit test  to check.
Now it's possible to say:

```
> bin/stansummary -f bern[1-4].csv -i theta
```
as well as 
```
> bin/stansummary -i theta -f bern[1-4].csv
```

this is ready for re-review.